Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
Putting this line back into the code seems to cause problems for me. I get the error 

    Parameter "id" for route "admin_caponica_foo_edit" must match "[^/]++" ("" given) 
    to generate a corresponding URL. 

After some digging around, I found that it was failling at this line of the code. The object already exists (it is being edited) and a call to `$values = $this->getIdentifierValues($entity);` works fine. 

However, for some possibly-doctrine-related reason that I don't fully understand, this condition (checking if the entity is in the unit of work) is returning false ... which leads to sonata trying to build an edit URL with a null id. Is it possible that it's getting a different EntityManager instance when using `$this->getEntityManager($entity)`? Or is there another reason it might not be seeing the (valid, existing) entity in the UOW?

If we look back at your case, @acrobat, is there any other way that you can test if your object is new instead of using this UOW check?